website/docs/guides/message-extraction.md

```diff
@@ -0,0 +1,186 @@
+
+This extractor uses the dependency tree of files, rather than just a glob pattern, to crawl imports and discover files more accurately.
+
+By doing so, it creates a more optimized catalog that only contains the messages needed for each page.
+
+The catalogs would still contain duplicating messages for common components, but it would be much better than the current approach.
+
+<img src="https://user-images.githubusercontent.com/1586852/221664457-24f1e9a5-5916-4a09-bdf1-2530831dafc4.jpg" />
+
+To start using `experimental-extractor`, you need to add the following section to lingui config:
+```ts
+/**
+ *
+ * @type {import('@lingui/conf').LinguiConfig}
+ */
+module.exports = {
+  // remove everethying from `catalogs` property
+  catalogs: [],
+  // highlight-start
+  experimental: {
+    extractor: {
+      // glob pattern of entrypoints
+      // this will find all nextjs pages
+      entries: [
+        "<rootDir>/src/pages/**/*.tsx"
+      ],
+      // output pattern, this instruct extractor where to store catalogs
+      // src/pages/faq.tsx -> src/pages/locales/faq/en.po
+      output: "<rootDir>/{entryDir}/locales/{entryName}/{locale}"
+    }
+  },
+  // highlight-end
+}
+```
+
+And then call in the terminal:
+
+```bash
+lingui extract-experimental
+```
+
+#### Notes
+It's important to note that the dependency tree crawling heavily relies on tree-shaking, which is a technique used by modern bundlers to eliminate unused code from the final bundle.
+
+While tree-shaking is generally effective, it might work differently than what the user expects.
+
+For example, consider the following code:
+
+```ts
+import { msg } from "@lingui/macro"
+
+export const species = {
+  Cardano: [
+    {
+      startsAt: 0,
+      name: msg`Ghost`,
+      icon: "Ghost",
+    },
+    {
+      startsAt: 0.000001,
+      name: msg`Plankton`,
+      icon: "Plankton",
+    }
+  ]
+}
+```
+
+On the surface, it may appear that this code can be safely removed from the final bundle if it's not used. However, the `msg` function call can potentially produce a side effect, preventing the bundler from removing the entire `species` object from the final bundle. As a result, messages defined in this snippet may be included in more catalogs than expected.
+
+To avoid this issue, one solution is to wrap the `species` object inside an Immediately Invoked Function Expression (IIFE) and add the `/* @__PURE__ */` annotation.
+
+By adding this annotation to the IIFE, we are telling the bundler that the entire `species` object can be safely removed if it is not used or exported elsewhere in the code.
+
+## Supported source types
+
+The extractor supports TypeScript, Flow and JavaScript (Stage 3) out of the box.
+
+f you use some experimental features (Stage 0 - Stage 2) or frameworks with custom syntax such as Vue.js or Svelte, you may want to implement your custom extractor.
+
+Visit [Advanced: Custom Extractor](/guides/custom-extractor.md) to learn how to create custom extractor.
```

Comment:
```suggestion
Visit [Advanced: Custom Extractor](/docs/guides/custom-extractor.md) to learn how to create a custom extractor.
```

Reply:
i copy-pasted if from somewhere. I don't understand how links should look like, because in on case you change it from way to another.